src/components/TablePopover/TableSearchPopover.tsx

```diff
@@ -12,36 +12,40 @@ import { useOnClickOutside } from "hooks";
 const { blue, gray } = palette;
 
 interface TableSearchPopoverProps {
-  value: string;
-  onChange: (search: string) => void;
-  onConfirm: () => void;
   "data-cy"?: string;
+  onConfirm: (search: string) => void;
   placeholder?: string;
+  value: string;
 }
 
 export const TableSearchPopover: React.FC<TableSearchPopoverProps> = ({
   "data-cy": dataCy,
-  onChange,
   onConfirm,
   placeholder,
   value,
 }) => {
+  const [input, setInput] = useState(value);
   const [active, setActive] = useState(false);
-  const iconColor = value === "" ? gray.dark2 : blue.light1;
+  const iconColor = value === "" ? gray.dark2 : blue.base;
 
   const buttonRef = useRef(null);
   const popoverRef = useRef(null);
 
   // Handle onClickOutside
   useOnClickOutside([buttonRef, popoverRef], () => setActive(false));
 
-  const closePopup = () => {
-    onConfirm();
+  // If the value from the URL has changed, update the input.
```

Comment:
I noticed that when changing the page size, filters and sorts are removed. Is that fixable? That may not be related to this `useEffect` but wanted to call it out.

Reply:
oops yeah seems to be a pre-existing bug since the other pages in Spruce also show this behavior! i think i fixed it